### PR TITLE
MAID 1694: net invariant output, refactor, doc

### PR DESCRIPTION
**Pull request by dhardy**
These are changes I've made while working on 1694, but not directly related to fixing it.

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by dhardy** [comment]
Travis test timed out. AppVeyer failed on `more_than_group_size_nodes` test, but it works for me.

File: src/ack_manager.rs [hunk 1]
```diff
@@ -41,10 +41,12 @@ pub struct AckManager {
     received: MessageFilter<Ack>,
 }
 
+/// An identifier for a waiting-to-be-acknowledged message (a hash of the message).
 #[derive(Clone, Copy, Eq, Hash, Ord, PartialEq, PartialOrd, RustcDecodable, RustcEncodable)]
 pub struct Ack(u64);
 
 impl AckManager {
+    /// Create a new manager, with empty lists.
```
Comment (afck) [inline src/ack_manager.rs]: Let's use the third person singular present indicative ("Creates") in method docs:
https://aturon.github.io/style/comments.html#sentence-structure
(I do like the idea of using questions for `-> bool` methods, though, if nobody objects.)

File: src/routing_table/network_tests.rs [hunk 2]
```diff
@@ -316,11 +316,13 @@ pub fn verify_network_invariant<'a, T, U>(nodes: U)
             }
             if prefix1.is_compatible(prefix2) {
                 panic!("Group prefixes should be disjoint, but these are not:\n\
-                    Group {:?}: {:?}\n\
-                    Group {:?}: {:?}",
+                    Group {:?}, according to node {:?}: {:?}\n\
+                    Group {:?}, according to node {:?}: {:?}",
```
Comment (afck) [inline src/routing_table/network_tests.rs]: Ah, yes, I already missed that bit from the logs yesterday! :+1:

Comment (dhardy) [inline src/routing_table/network_tests.rs]: This is already my second implementation of this extension. I had nothing to test with when I first wrote this output.

File: src/ack_manager.rs [hunk 3]
```diff
@@ -54,16 +56,20 @@ impl AckManager {
         }
     }
 
-    // Handle received ack.
+    // Handle received ack (remove the corresponding message from the list of
```
Comment (afck) [inline src/ack_manager.rs]: That should be `Handles` (third person etc.) - also in the other comments.

File: src/states/node.rs [hunk 4]
```diff
@@ -1344,15 +1332,13 @@ impl Node {
     }
 
     fn connect(&mut self,
-               encrypted_connection_info: Vec<u8>,
-               nonce_bytes: [u8; box_::NONCEBYTES],
-               their_public_id: PublicId,
+               conn_into: ConnectionInfo,
```
Comment (Fraser999) [inline src/states/node.rs]: Typo in `conn_into`?

**Pull request merged**